sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -1017,26 +988,25 @@ int gcoap_get_resource_list(void *buf, size_t maxlen, uint8_t cf)
 
 int gcoap_add_qstring(coap_pkt_t *pdu, const char *key, const char *val)
 {
-    size_t qs_len = strlen((char *)pdu->qs);
+    char qs[NANOCOAP_URI_MAX];
```

Comment:
NANOCOAP_QS_MAX definition goes away in nanocoap.h.

Reply:
Why not keep `NANOCOAP_QS_MAX` and use it here? To me it makes sense to separate these two defines, even if they have the same value.

Reply:
I am inclined to leave as-is until we have a reason to differentiate max length of query string. Why add a constant we don't need? The use is internal/invisible to the user, so we can add it whenever necessary.

Reply:
To me it feels as if the define is forcefully removed with this PR while there was still a use case for the define. For me defines are a way to make code more readable by giving a meaning to a constant. 

Using the `NANOCOAP_URI_MAX` define here where the use case is not to fix the size of a uri array feels dirty. 

So my opinion would be to keep the `NANOCOAP_QS_MAX`, maybe define the value as `NANOCOAP_URI_MAX`, but to at least keep it here for clarity reasons.

Reply:
I understand your motivation. I removed the macro because nanocoap removed it from generic use in #8772, and left if for gcoap for historical reasons. I just pushed a fixup for gcoap only, as it was before this PR. We can reinstate NANOCOAP_QS_MAX generically for nanocoap when we work on merging #8920.